rust/src/detect/mod.rs

```diff
@@ -25,3 +25,19 @@ pub mod stream_size;
 pub mod uint;
 pub mod uri;
 pub mod requires;
+
+/// Enum trait that will be implemented on enums that
+/// derive StringEnum.
+pub trait Enum<T> {
```

Comment:
Can we add a test for this trait and derive macro?  I know we don't have them for `AppLayerEvent` and `AppLayerFrameType`, but in the position of not being the author, I can see how they would be really useful.

Reply:
Right